src/crypto/CryptState.h

```diff
@@ -13,14 +13,11 @@ class CryptState {
 private:
 	Q_DISABLE_COPY(CryptState)
 public:
-	static constexpr unsigned int IV_LENGTH  = 0;
-	static constexpr unsigned int KEY_LENGTH = 0;
-
 	/// The plain input and encrypted output of the block cipher always preserve the same length.
 	/// However, in order to transmit the authenticate tag and the packet sequence number, an extra
 	/// "head" is needed to be attached to the packet.
-	/// This length varies from cipher to cipher and is stored in this `HEAD_LENGTH`.
-	static const unsigned int HEAD_LENGTH = 0;
+	/// This length varies from cipher to cipher and is stored in this `headLength`.
+	unsigned int headLength = 0;
```

Comment:
This is a very tricky thing. If I keep it as a constant, when accessing `csCrypt->HEAD_LENGTH`, it will always give me zero, no matter how I try to override it later.

Reply:
> no matter how I try to override it later

Well without using const-casts you shouldn't be able to override it in the first place without the compiler getting angry at you xD
And with const-casts the compiler lets you overwrite the value at runtime but probably inlines the value at compile-time since it is declared as constant (and at compile-time the compiler sees that this is set to zero, so that's what it is going to inline). At least that is what I think is happening here

Reply:
If we really want to save some time getting this `headLength`, I think we'd better store it into the `Connection` object instead of accessing it every time. As long as the type of cipher is not determined at compile-time, we can't really count on this value-accessing being optimized. :/

Reply:
Yes absolutely. I was not trying to criticize this change but merely wanted to explain the behavior you were seeing :)

Reply:
No worries, I got your idea :)